File: src/main/java/com/seedsanskriti/service/WishlistService.java
package com.seedsanskriti.service;

import java.util.List;

import org.springframework.security.core.Authentication;

import com.seedsanskriti.dto.WishlistItemResponse;

public interface WishlistService {

    String addToWishlist(Long productId, Authentication authentication);

    List<WishlistItemResponse> getMyWishlist(Authentication authentication);

    String removeFromWishlist(Long productId, Authentication authentication);
}

File: seedsanskriti_frontend/src/services/wishlistService.js
import apiClient from './apiClient';

const wishlistService = {
  getMyWishlist: () => apiClient.get('/wishlist').then((res) => res.data),

  addToWishlist: (productId) =>
    apiClient.post(`/wishlist/${productId}`).then((res) => res.data),

  removeFromWishlist: (productId) =>
    apiClient.delete(`/wishlist/${productId}`).then((res) => res.data),
};

export default wishlistService;
